clog/builder.go

```diff
@@ -31,6 +31,7 @@ type builder struct {
 	labels          map[string]struct{}
 	comments        map[string]struct{}
 	skipCallerJumps int
+	settings        Settings
```

Comment:
this feels like it's getting a little chunky since each call to `clog.Ctx()` will essentially create one of these right? Should we start thinking about things like storing a pointer instead?

Reply:
A pointer to the builder, or to the settings?

Reply:
pointer to settings. I assume a pointer to the builder is needed in some respects anyway to store information (i.e. modify state of the builder in some calls)? I haven't read through the code though so I could be incorrect there

Reply:
aye, correct.  Settings structs are lightweight, and builders are short lived.  I'm not particularly concerned about carrying over the details at this moment.  If it becomes a concern I'd rather reduce the carryover to the required properties, such as the logging level.